src/client.rs

```diff
@@ -439,6 +458,50 @@ where
 
     /// Returns `true` if an update occurred and `false` otherwise.
     async fn update_root(&mut self) -> Result<bool> {
+        let updated = Self::update_root_with_repos(
+            &self.config,
+            &mut self.tuf,
+            Some(&self.local),
+            &self.remote,
+        )
+        .await?;
+
+        /////////////////////////////////////////
+        // TUF-1.0.9 §5.1.9:
+        //
+        //     Check for a freeze attack. The latest known time MUST be lower than the expiration
+        //     timestamp in the trusted root metadata file (version N). If the trusted root
+        //     metadata file has expired, abort the update cycle, report the potential freeze
+        //     attack. On the next update cycle, begin at step 5.0 and version N of the root
+        //     metadata file.
+
+        // TODO: Consider moving the root metadata expiration check into `tuf::Tuf`, since that's
+        // where we check timestamp/snapshot/targets/delegations for expiration.
+        if self.tuf.trusted_root().expires() <= &Utc::now() {
+            error!("Root metadata expired, potential freeze attack");
```

Comment:
not for this cl, but maybe it would be better to have a separate error variant specifically for this type of problem